es/clientsigner.go

```diff
@@ -0,0 +1,62 @@
+//   Copyright 2017 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package es
+
+import (
+	"errors"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/signer/v4"
+	"github.com/sha1sum/aws_signing_client"
+	"gopkg.in/olivere/elastic.v5"
+	"strings"
+)
+
+func checkError(splittedEndpoint []string, creds *credentials.Credentials) error {
+	if creds == nil {
```

Comment:
Do credentials still work without being a pointer ? Are they small in memory ? If both statements are true, perhaps you should not be using a pointer, it complicate the code with this sort of checks. Or you can assume it is the caller’s responsibility to provide a non-nil pointer.

Reply:
According to http://docs.aws.amazon.com/sdk-for-go/api/aws/credentials, all functions allowing to get a credentials return a pointer. That's why I use pointers.